akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -929,6 +929,15 @@ final class Source[Out, Mat](delegate: scaladsl.Source[Out, Mat]) extends Graph[
   def toMat[M, M2](sink: Graph[SinkShape[Out], M], combine: function.Function2[Mat, M, M2]): javadsl.RunnableGraph[M2] =
     RunnableGraph.fromGraph(delegate.toMat(sink)(combinerToScala(combine)))
 
+  /**
+   * Connect this `Source` to the `Sink.ignore` and run it. Elements from the stream will be consumed and discarded.
+   *
+   * Note that the `ActorSystem` can be used as the implicit `materializer` parameter to use the
+   * [[akka.stream.SystemMaterializer]] for running the stream.
+   */
+  def run()(implicit materializer: Materializer): Future[Done] =
```

Comment:
We can remove the implicit marker from the javadsl. We should also remove it from the javadoc. It makes no sense for java users.

Reply:
Do we have tests written in Java to prove that the API usable from Java? If so, we should certainly update it.

Reply:
The Java API should have one overload that takes `Materializer` and one that takes a `ClassicActorSystemProvider` and picks up the system materializer from inside, look at the other preexisting `run` methods in the Java DSL.

It should also be a single parameter list.

Reply:
(sorry for missing this in my last review)

Reply:
and it should return CompletionStage, not Future, in javadsl

Reply:
oh, I missed that one. CompletionStage, of course.

Reply:
I am sorry for this oversight. I fixed all these issues.